### PR TITLE
Arm migration docs and template enhancements

### PR DESCRIPTION
**Pull request by markcowl**
Fixes for 
- [x] https://github.com/Azure/typespec-azure-pr/issues/3842
- [x] https://github.com/Azure/typespec-azure/issues/88
- [x] https://github.com/Azure/typespec-azure-pr/issues/3466

### CODE REVIEW

**Review by timotheeguerin — APPROVED** [review]

**Review by allenjzhang — APPROVED** [review]
nit: break it out into individual sections. Can be done later.

File: docs/howtos/migrate-swagger/migrate-arm-tips.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+```tsp
+interface Widgets {
+  createOrUpdate is ArmResourceCreateOrReplaceAsync<Widget>;
+}
+```
+
+The name of the request body parameter is `resource` so you can change the name in clients using an augment decorator
+
+```tsp
+@@projectedName(Widgets.createOrUpdate::parameters.resource,
```
Comment (allenjzhang) [inline docs/howtos/migrate-swagger/migrate-arm-tips.md:29]: @projectedName?

File: docs/howtos/migrate-swagger/migrate-arm-tips.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+listBySubscription is ArmListBySubscription<Widget, Error = MyCustomError>;
+```
+
+### Converting Synchronous Operations to LROs
+
+You can generally choose an asynchronous operation template that matches your operation.
+
+#### Templates for Async PUT Operations
+
+- `ArmCreateOrReplaceAsync` is a PUT operation that uses the 'resource' definition in the request body, and return a `200` response and a `201` response, both of which contain the created/updated resource in the response payload. The 201 response contains 'Location` LRO header.
```
Comment (allenjzhang) [inline docs/howtos/migrate-swagger/migrate-arm-tips.md:115]: correction needed:  ArmResourceCreateOrReplaceSync

Comment (markcowl) [reply docs/howtos/migrate-swagger/migrate-arm-tips.md:115]: No, this is describing both of the async operations

File: docs/howtos/migrate-swagger/migrate-arm-tips.md (hunk 1 above)
Comment (markcowl) [reply docs/howtos/migrate-swagger/migrate-arm-tips.md:29]: Yes, until autorest supports clientName

**Pull request merged**